vald/evm/evm.go

```diff
@@ -262,79 +264,49 @@ func (mgr Mgr) ProcessGatewayTxConfirmation(event *types.ConfirmGatewayTxStarted
 		return err
 	}
 
-	var events []types.Event
-	for i, txlog := range txReceipt.Logs {
-		if !bytes.Equal(event.GatewayAddress.Bytes(), txlog.Address.Bytes()) {
-			continue
-		}
-
-		switch txlog.Topics[0] {
-		case ContractCallSig:
-			gatewayEvent, err := DecodeEventContractCall(txlog)
-			if err != nil {
-				mgr.logger().Debug(sdkerrors.Wrap(err, "decode event ContractCall failed").Error())
-				continue
-			}
+	events := mgr.processGatewayTxLogs(event.Chain, event.GatewayAddress, txReceipt.Logs)
+	mgr.logger().Infof("broadcasting vote %v for poll %s", events, event.PollID.String())
+	_, err = mgr.broadcaster.Broadcast(context.TODO(), voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain, events...)))
 
-			if err := gatewayEvent.ValidateBasic(); err != nil {
-				mgr.logger().Debug(sdkerrors.Wrap(err, "invalid event ContractCall").Error())
-				continue
-			}
+	return err
+}
 
-			events = append(events, types.Event{
-				Chain: event.Chain,
-				TxID:  event.TxID,
-				Index: uint64(i),
-				Event: &types.Event_ContractCall{
-					ContractCall: &gatewayEvent,
-				},
-			})
-		case ContractCallWithTokenSig:
-			gatewayEvent, err := DecodeEventContractCallWithToken(txlog)
-			if err != nil {
-				mgr.logger().Debug(sdkerrors.Wrap(err, "decode event ContractCallWithToken failed").Error())
-				continue
-			}
+// ProcessMultipleGatewayTxConfirmations votes on the correctness of an EVM chain multiple gateway transactions
+func (mgr Mgr) ProcessMultipleGatewayTxConfirmations(event *types.ConfirmGatewayTxsStarted) error {
```

Comment:
@cgorenflo do you have strong opinion on the naming? or we can also change the event name

Reply:
We can also rename the old functions to `ProcessSingleGatewayTxConfirmation` if you like that better. I just think because the name is so long and the difference is only a single character it makes reading the code much harder.